lib/services/platform-environment-requirements.ts

```diff
@@ -0,0 +1,135 @@
+
+			const selectedOption = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_MESSAGE, [
+				PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+				PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME,
+				PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME,
+				PlatformEnvironmentRequirements.BOTH_CLOUD_BUILDS_AND_SETUP_SCRIPT_OPTION_NAME
+			]);
+
+			if (selectedOption === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
+				await this.processCloudBuilds(platform);
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME) {
+				await this.$doctorService.runSetupScript();
+
+				if (await this.$doctorService.canExecuteLocalBuild(platform)) {
+					return true;
+				}
+
+				const option = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_AFTER_SETUP_SCRIPT_MESSAGE, [
+					PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+					PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME
+				]);
+
+				if (option === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
+					await this.processCloudBuilds(platform);
+				}
+
+				if (option === PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME) {
+					this.processManuallySetup(platform);
+				}
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME) {
+				this.processManuallySetup(platform);
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.BOTH_CLOUD_BUILDS_AND_SETUP_SCRIPT_OPTION_NAME) {
+				await this.processBothCloudBuildsAndSetupScript(platform);
+				if (await this.$doctorService.canExecuteLocalBuild(platform)) {
+					return true;
+				}
+
+				this.processManuallySetup(platform);
+			}
+		}
+
+		return true;
+	}
+
+	private async processCloudBuilds(platform: string): Promise<void> {
+		await this.processCloudBuildsCore(platform);
+		this.fail(this.getCloudBuildsMessage(platform));
+	}
+
+	private async processCloudBuildsCore(platform: string): Promise<IExtensionData> {
+		const installedExtensions = this.$extensibilityService.getInstalledExtensions();
+		if (!installedExtensions[PlatformEnvironmentRequirements.NATIVESCRIPT_CLOUD_EXTENSION_NAME]) {
+			return this.$extensibilityService.installExtension(PlatformEnvironmentRequirements.NATIVESCRIPT_CLOUD_EXTENSION_NAME);
+		}
+	}
+
+	private	getCloudBuildsMessage(platform: string): string {
+		const cloudCommandName = this.cliCommandToCloudCommandName[this.$commandsService.currentCommandData.commandName];
+		if (!cloudCommandName) {
+			return `In order to test your application use the $ tns login command to log in with your account and then $ tns cloud build command to build your app in the cloud.`;
+		}
+
+		if (!platform) {
+			return `Use the $ tns login command to log in with your account and then $ tns cloud ${cloudCommandName.toLowerCase()} command to build your app in the cloud.`;
```

Comment:
Doesn't `cloudCommandName` hold the full command - `tns cloud <operation>`?
This means that if the user was running `build` for example we'd print 
`... and then $ tns cloud tns cloud build command ... `

Reply:
On a side note I think it may need a better string for this, because we tell the user to execute `cloudCommandName` to **build the app in the cloud**.
The user may have executed `run` and if this is the case we'd suggest `tns cloud run` to **build the app**.